core-web/libs/sdk/analytics/src/lib/core/shared/models/event.model.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * Event models for DotCMS Analytics
+ * Contains interfaces for different types of analytics events
+ */
+
+import { DotCMSDeviceData, DotCMSPageData, DotCMSUtmData } from './data.model';
+
+import {
+    CustomEventType,
+    DotCMSEventType,
+    DotCMSPredefinedEventType
+} from '../constants/dot-content-analytics.constants';
+
+/**
+ * Base structure for all analytics events.
+ * All events share this common structure.
+ */
+export interface DotCMSEventBase {
+    /** The type of event being tracked */
+    event_type: DotCMSEventType;
+    /** Local timestamp when the event occurred */
+    local_time: string;
+}
+
+/**
+ * Pageview-specific analytics event structure.
+ * Contains data specific to page view tracking.
+ */
+export interface DotCMSPageViewEvent extends DotCMSEventBase {
+    event_type: typeof DotCMSPredefinedEventType.PAGEVIEW;
+    /** Pageview-specific event data with structured format */
+    data: {
+        /** Page data associated with the event */
+        page: DotCMSPageData;
+        /** Device and browser information */
+        device: DotCMSDeviceData;
+        /** UTM parameters for campaign tracking (optional) */
+        utm?: DotCMSUtmData;
+        /** Custom data associated with the event */
+        custom?: Record<string, unknown>;
```

Comment:
Does pageview can have custom props? and if so... why id not `data.custom` like the `DotCMSCustomEvent`?